### DISCUSSION TITLE
Add a "From zero to hero" contributor roadmap article to https://pulsar.apache.org/contribute/contributor-roadmap

### DISCUSSION THREAD

**Discussion by visortelle** [apache/pulsar#22176] [Ideas]
Imagine that you are a Pulsar user or interested in data streaming in general and would like to contribute to some related open-source project.

You have some prior programming experience but you've worked on other kinds of projects and never developed distributed streaming systems.

The current documentation doesn't describe the prior knowledge list recommended to contribute to Pulsar.

Having a clear roadmap of what to learn and where to start would bring more contributors to the project.

It may have a similar format:

- [ ] Java
  **Recommended resources:**
  ...
- [ ] Concurrency and Parallelism
  **Recommended resources:**
  ...
- [ ] Networking
  **Recommended resources:**
  ...
- [ ] Data streaming
  **Recommended resources:**
  ...
- [ ] Cyber-security
  **Recommended resources:**
  ...
- [ ] Storage and Bookkeeper
  **Recommended resources:**
  ...
- [ ] Pulsar internals
  **Recommended resources:**
  ...
- [ ] Important libraries to know
  - [ ] Netty
  - [ ] Jetty
  - [ ] Lombok
  ...
- [ ] Issue list for starting contributing
  ...

Of course, it takes time to learn these things and can add some additional load on the PR reviewers' shoulders at first. But it also could help to grow a new wave of contributors who are loyal to the project from the beginning, which is important for growing the Pulsar contributors community.

Mailing list thread: https://lists.apache.org/thread/99kych4r4lcw33qs49bwk9zxmss8vggl

**Comment by asafm** [comment]
Where would you like to place this?

**Reply by visortelle** [reply]
Probably Contribute -> Development -> Developer Roadmap

![IMG_0472](https://github.com/apache/pulsar/assets/9302460/08056280-847d-4db2-9b61-cdd21c1ec39f)

**Reply by asafm** [reply]
Cool.
Maybe "Prerequisite Knowledge"? or "Recommended Knowledge"?

**Reply by visortelle** [reply]
I think “Recommended <something>” fits better here.

Therefore “Recommended Knowledge” until we haven't better options.
